meteor/client/ui/Shelf/BucketPanel.tsx

```diff
@@ -919,3 +944,81 @@ export const BucketPanel = translateWithTracker<Translated<IBucketPanelProps>, I
 		)
 	)
 )
+
+interface DropzoneHolderProps {
+	id: string
+	bucketId: BucketId
+	url: string
+	hidden: boolean
+	t: TFunction<'translation', undefined>
+	showStyleBaseId: ShowStyleBaseId
+}
+const DropzoneHolder = (props: DropzoneHolderProps) => {
+	let frame: HTMLIFrameElement | null = null
+	let mounted = false
+
+	const setDropzoneElement = (frameRef: HTMLIFrameElement | null) => {
+		frame = frameRef
+		if (frame && !mounted) {
+			registerHandlers()
+			mounted = true
+		} else {
+			unregisterHandlers()
+			mounted = false
+		}
+	}
```

Comment:
I think component state in React should only really be held and mutated via the `React.useState` hook. I feel like this way of doing it can very quickly end up in some undefined situations, if there are spontaneous component re-renders. The `mounted` will be false then, but the `setDropzoneElement` will not re-run, because as far as react will be concerned, the reference has already been set.

See a note below on how to solve this.